plugin/pkg/shoot/dns/admission.go

```diff
@@ -149,56 +151,59 @@ func (d *DNS) Admit(a admission.Attributes, o admission.ObjectInterfaces) error
 // assignDefaultDomainIfNeeded generates a domain <shoot-name>.<project-name>.<default-domain>
 // and sets it in the shoot resource in the `spec.dns.domain` field.
 // If for any reason no domain can be generated, no domain is assigned to the Shoot.
-func assignDefaultDomainIfNeeded(shoot *garden.Shoot, projectLister gardenlisters.ProjectLister, secretLister kubecorev1listers.SecretLister) {
+func assignDefaultDomainIfNeeded(shoot *garden.Shoot, projectLister gardenlisters.ProjectLister, secretLister kubecorev1listers.SecretLister) error {
 	project, err := admissionutils.GetProject(shoot.Namespace, projectLister)
 	if err != nil {
-		return
+		return apierrors.NewInternalError(err)
 	}
 
 	var domainSecrets []*corev1.Secret
 	deprecatedSelector, err := labels.Parse(fmt.Sprintf("%s=%s", v1alpha1constants.DeprecatedGardenRole, common.GardenRoleDefaultDomain))
 	if err != nil {
-		return
+		return apierrors.NewInternalError(err)
 	}
 	secrets, err := secretLister.Secrets(v1alpha1constants.GardenNamespace).List(deprecatedSelector)
 	if err != nil {
-		return
+		return apierrors.NewInternalError(err)
 	}
 	domainSecrets = append(domainSecrets, secrets...)
 
 	selector, err := labels.Parse(fmt.Sprintf("%s=%s", v1alpha1constants.GardenRole, common.GardenRoleDefaultDomain))
 	if err != nil {
-		return
+		return apierrors.NewInternalError(err)
 	}
 	secrets, err = secretLister.Secrets(v1alpha1constants.GardenNamespace).List(selector)
 	if err != nil {
-		return
+		return apierrors.NewInternalError(err)
 	}
 	domainSecrets = append(domainSecrets, secrets...)
 
 	if len(domainSecrets) == 0 {
-		return
+		return nil
 	}
 
 	var shootDomain *string
 	if shoot.Spec.DNS != nil {
 		shootDomain = shoot.Spec.DNS.Domain
 	}
 
-	if shootDomain != nil {
-		return
-	}
-
 	for _, secret := range domainSecrets {
 		_, domain, _, _, err := common.GetDomainInfoFromAnnotations(secret.Annotations)
 		if err != nil {
-			return
+			return apierrors.NewInternalError(err)
 		}
 
 		if shootDomain != nil && strings.HasSuffix(*shootDomain, domain) {
```

Comment:
```suggestion
		if shootDomain != nil && strings.HasSuffix(*shootDomain, "." + domain) {
```
Not really related to this PR, but I think this check should be improved
At the moment you can specify something like `foo<default domain>` (notice: **no** dot seperator), which might be intended and then your Providers section will be set to nil by the admission plugin.
Is this the wanted behaviour?

Reply:
No, valid point, nice catch. Fixed, PTAL again! :)